db/blob_log_format.cc

```diff
@@ -0,0 +1,245 @@
+#include "db/blob_log_format.h"
+#include "util/coding.h"
+#include "util/crc32c.h"
+
+
+namespace rocksdb {
+namespace blob_log {
+
+
+BlobLogHeader::BlobLogHeader()
+: magic_number_(kMagicNumber),
+  has_ttl_(false),
+  has_ts_(false)
+{
+}
+
+BlobLogFooter::BlobLogFooter()
+: magic_number_(kMagicNumber),
+  blob_count_(0),
+  has_ttl_(false), has_ts_(false)
+{
+}
+
+Status BlobLogFooter::DecodeFrom(Slice* input)
+{
+  Status s;
+  uint32_t offset = 0;
+
+  const char *ptr = input->data();
+  uint32_t mn = DecodeFixed32(ptr);
+  if (mn != kMagicNumber)
+  {
+    return Status::Corruption("bad magic number");
+  }
+
+  magic_number_ = mn;
+
+  offset += sizeof(uint32_t);
+  uint32_t val = (DecodeFixed32(ptr + offset) >> 8);
+  has_ttl_ = has_ts_ = false;
+  RecordSubType st = (RecordSubType)val;
+  if (st == kTTLType) {
+    has_ttl_= true;
+  } else if (st == kTimestampType) {
+    has_ts_ = true;
+  } else {
+    return Status::Corruption("bad sub type");
+  }
+
+  offset += sizeof(uint32_t);
+  blob_count_ = DecodeFixed64(ptr + offset);
+
+  offset += sizeof(uint64_t);
```

Comment:
GetFixed64(Slice*, uint64*) will automatically remove the decoded part in the Slice. It will make code easier.